bundle/Service/ScheduledVisibilityService.php

```diff
@@ -0,0 +1,141 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Netgen\Bundle\IbexaScheduledVisibilityBundle\Service;
+
+use DateTime;
+use Ibexa\Contracts\Core\Repository\Values\Content\Content;
+use Ibexa\Contracts\Core\Repository\Values\Content\Field;
+use Ibexa\Core\FieldType\Date\Value as DateValue;
+use Ibexa\Core\FieldType\DateAndTime\Value as DateAndTimeValue;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Configuration\ScheduledVisibilityConfiguration;
+use Netgen\Bundle\IbexaScheduledVisibilityBundle\Enums\VisibilityUpdateResult;
+use OutOfBoundsException;
+
+use function in_array;
+
+final class ScheduledVisibilityService
+{
+    public function __construct(
+        private readonly ScheduledVisibilityConfiguration $configurationService,
+    ) {}
+
+    /**
+     * @throws OutOfBoundsException
+     */
+    public function updateVisibilityIfNeeded(Content $content): VisibilityUpdateResult
+    {
+        if (!$this->accept($content)) {
+            return VisibilityUpdateResult::NoChange;
+        }
+
+        $handler = $this->configurationService->getHandler();
+
+        if ($this->shouldBeHidden($content) && !$handler->isHidden($content)) {
+            $handler->hide($content);
+
+            return VisibilityUpdateResult::Hidden;
+        }
+
+        if ($this->shouldBeVisible($content) && $handler->isHidden($content)) {
+            $handler->reveal($content);
+
+            return VisibilityUpdateResult::Revealed;
+        }
+
+        return VisibilityUpdateResult::NoChange;
+    }
+
+    public function accept(Content $content): bool
+    {
+        if (!$this->configurationService->isEnabled()) {
```

Comment:
Move the logic for enabled state and content types to the consuming code - to the command and to the event subscriber.

Reply:
fixed here 4fd334ca04b686baefc4cbc1f1d9dea504bfeb50